src/target/riscv/riscv.c

```diff
@@ -5029,6 +5040,73 @@ int riscv_dmi_write_u64_bits(struct target *target)
 	return r->dmi_write_u64_bits(target);
 }
 
+static int check_if_trigger_is_availible(struct target *target, unsigned int t)
+{
+	/* If we can't write tselect, then this hart does not support triggers. */
+	if (riscv_set_register(target, GDB_REGNO_TSELECT, t) != ERROR_OK)
+		return ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
+	riscv_reg_t tselect_rb;
+	if (riscv_get_register(target, &tselect_rb, GDB_REGNO_TSELECT) != ERROR_OK)
+		return ERROR_FAIL;
+	/* Mask off the top bit, which is used as tdrmode in old
+	 * implementations. */
+	tselect_rb &= ~(1ULL << (riscv_xlen(target) - 1));
+	if (tselect_rb != t)
+		return ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
+	return ERROR_OK;
+}
+
+static int get_trigger_types(struct target *target, unsigned int *trigger_tinfo,
+		riscv_reg_t tdata1)
+{
+	riscv_reg_t tinfo;
+	const bool may_support_multiple_trigger_types =
```

Comment:
If you like, this can be called simply `tinfo_exists` or `tinfo_readable`, which is much shorter and still very clear.

```suggestion
	const bool tinfo_exists =
```

Reply:
Addressed. This variable is used only once, so I decided to just propagate the definition.

Reply:
@en-sc Can you please check if the change has been pushed? I don't see it in the diff on github.

Reply:
Please, see [the diff](https://github.com/riscv/riscv-openocd/compare/d4cdb997e455801e1c63572f113fc600e8d0fad2..ea43a461a42681651f30d07b3e8ff03457ac1f70#diff-b4aa16f9e42cb8f0934baa7c8e0ec9c70a369bef98b99b26ae2e896c8aa95d6aR5064). I did forget to push the changes. Sorry.

Reply:
No problem - thanks.